sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
Yeah, I agreed. With current behavior, it is unavoidable to have some strange queries with `_corrupted_record`.

I'd suggest as https://github.com/apache/spark/pull/18865#discussion_r132688380, we should document `_corrupted_record` in CSV, JSON data sources is a derived column and can be incorrect if not used with other columns.